src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs

```diff
@@ -819,30 +819,10 @@ public static async Task<IEnumerable<IStorageItemWithPath>> GetDraggedStorageIte
 								}
 							}
 
-							// get storage items from path
-							var storageItems = await Task.WhenAll(
-								itemPaths.Select<string, Task<IStorageItem?>>(async path =>
-								{
-									try
-									{
-										if (NativeFileOperationsHelper.HasFileAttribute(path, FileAttributes.Directory))
-											return await StorageFolder.GetFolderFromPathAsync(path);
-										else
-											return await StorageFile.GetFileFromPathAsync(path);
-									}
-									catch
-									{
-										return null;
-									}
-								})
-							);
-
-							foreach (var item in storageItems)
+							foreach (var path in itemPaths)
 							{
-								if (item is null)
-									continue;
-
-								itemsList.Add(item.FromStorageItem());
+								var isDirectory = NativeFileOperationsHelper.HasFileAttribute(path, FileAttributes.Directory);
+								itemsList.Add(StorageHelpers.FromPathAndType(path, isDirectory ? FilesystemItemType.Directory : FilesystemItemType.File));
```

Comment:
nice! was looking for something like this.
just wasn't sure if the code paths actually needed the `IStorageItem` at all

Reply:
IStorageItem only needed if you don't have the item's path / for locations where StorageFile.GetFileFromPathAsync() dies not work (MTP/phone storage). 👍